pkg/cloudprovider/aws/errors.go

```diff
@@ -43,6 +43,10 @@ var (
 
 type SpotFallbackError error
 
+type InstanceTerminatedError struct {
```

Comment:
can you just make this an error instead of a struct like `SpotFallbackError`?

Reply:
I actually started with that, but I think the `InstanceTerminatedError ` error ends up being the `*errors.errorString` type.  Which means that the `IsTerminatedError` would return true for errors when we expect them to return false.  Now that I think of it, I'm not 100% sure the `isSpotFallback` method is doing what we intend either.  Will have to test it further.

Here's an example, btw.
https://go.dev/play/p/cswSjwGQFL2

Reply:
o wow, good catch!